compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -438,7 +439,16 @@ fn link_rlib<'a>(
     // Add all bundled static native library dependencies.
     // Archives added to the end of .rlib archive, see comment above for the reason.
     for lib in packed_bundled_libs {
-        ab.add_file(&lib)
+        let mut src = fs::File::open(&lib).unwrap();
+        let mut x = Default::default();
+        src.read_to_end(&mut x).unwrap();
+        let (data, _) = create_wrapper_file(sess, b".rlib_entry".to_vec(), &x);
+        ab.add_file(&emit_wrapper_file(
+            sess,
+            &data,
+            tmpdir,
+            lib.file_name().unwrap().to_str().unwrap(),
```

Comment:
Could you push this name to the `packed_bundled_libs` in advance to avoid retrieving it for the second time?